readthedocs/core/utils/__init__.py

```diff
@@ -133,7 +133,10 @@ def prepare_build(
 
     if build:
         # Send pending Build Status using Git Status API for External Builds.
-        send_external_build_status(version=version, build_pk=build.id, status=BUILD_STATUS_PENDING)
+        send_external_build_status(
+            version_type=version.type, build_pk=build.id,
+            commit=version.identifier, status=BUILD_STATUS_PENDING
```

Comment:
I feel like this isn't quite right. We're still just getting the commit _from_ the VCS that is getting checked out. This is still coming from the `Version` object. We also don't have the VCS-based commit when we trigger the build.

We need to pass this `commit` along to the actual build, so that we're triggering the specific build with this commit, not the random commit that's stored in the build object.